library/L1_Peripheral/stm32f10x/test/pwm_test.cpp

```diff
@@ -0,0 +1,161 @@
+#include "L1_Peripheral/stm32f10x/pwm.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+
+namespace sjsu::stm32f10x
+{
+EMIT_ALL_METHODS(Pwm);
+TEST_CASE("Testing stm32f10x PWM")
+{
+  static constexpr units::frequency::hertz_t kDummyClockRate = 8_MHz;
+  Mock<SystemController> mock_controller;
+  Mock<sjsu::Pin> mock_pwm_pin;
+
+  Fake(Method(mock_controller, PowerUpPeripheral));
+  Fake(Method(mock_pwm_pin, ConfigureFunction));
+  When(Method(mock_controller, GetClockRate)).AlwaysReturn(kDummyClockRate);
+  SystemController::SetPlatformController(&mock_controller.get());
+
+  TIM_TypeDef local_timer;
+
+  testing::ClearStructure(&local_timer);
+
+  Pwm::PwmPin_t mock_pwm = {
+    .pin       = mock_pwm_pin.get(),
+    .registers = &local_timer,
+    .channel   = 1,
+    .id        = SystemController::Peripherals::kTimer1
+  };
+
+  Pwm test_subject(mock_pwm);
+
+  SECTION("Initialize")
+  {
+    // Exercise
+    test_subject.Initialize();
+
+    // Verify
+    Verify(Method(mock_controller, PowerUpPeripheral).Using(mock_pwm.id))
+      .Once();
+  }
+
+  SECTION("ConfigureFrequency()")
+  {
+    // Setup
+    units::frequency::hertz_t frequency;
+
+    SUBCASE("Frequency 1")
+    {
+      frequency = 1_Hz;
+    }
+
+    SUBCASE("Frequency 2")
+    {
+      frequency = 100_Hz;
+    }
+
+    SUBCASE("Frequency 3")
+    {
+      frequency = 1_kHz;
+    }
+
+    SUBCASE("Frequency 4")
+    {
+      frequency = 50_Hz;
+    }
+
+    SUBCASE("Frequency 5")
+    {
+      frequency = 100_kHz;
+    }
+
+    SUBCASE("Frequency 6")
+    {
+      frequency = 1_MHz;
+    }
+
+    SUBCASE("Frequency 7")
+    {
+      frequency = 5_MHz;
+    }
+
+    INFO("Failure for frequency " << frequency);
+
+    // Calculate prescalar and divider
+    auto & system                   = SystemController::GetPlatformController();
+    const auto kPeripheralFrequency = system.GetClockRate(mock_pwm.id);
+    auto period = 1 / frequency;
+
+    uint32_t product = period * kPeripheralFrequency;
+    uint16_t k_max16_bits = ~0;
+
+    uint16_t prescalar = 0;
+    uint32_t divider;
+
+    do
+    {
+      prescalar++;
+      divider = product / prescalar;
+    } while (divider > k_max16_bits);
+
+    prescalar--;
+
+    // Exercise
+    test_subject.ConfigureFrequency(frequency);
+
+    // Verify
+    CHECK(prescalar == local_timer.PSC);
+    CHECK(divider == local_timer.ARR);
+  }
+
+  SECTION("SetDutyCycle()")
+  {
+    float duty_cycle;
```

Comment:
```suggestion
    // Setup
    float duty_cycle;
```

NIT: Add `// Setup` here.